sentinelhub/api/batch/statistical.py

```diff
@@ -0,0 +1,215 @@
+"""
+Module implementing an interface with
+`Sentinel Hub Batch Processing API <https://docs.sentinel-hub.com/api/latest/api/batch-statistical/>`__.
+"""
+import datetime as dt
+import logging
+import sys
+from dataclasses import dataclass, field
+from typing import Any, List, Optional, Union
+
+from dataclasses_json import CatchAll, LetterCase, Undefined
+from dataclasses_json import config as dataclass_config
+from dataclasses_json import dataclass_json
+
+from ...type_utils import Json, JsonDict
+from ..base_request import InputDataDict
+from ..statistical import SentinelHubStatistical
+from ..utils import datetime_config, enum_config, remove_undefined
+from .base import BaseBatchClient, BaseBatchRequest, BatchRequestStatus, BatchUserAction
+
+if sys.version_info < (3, 11):
+    from typing_extensions import NotRequired, TypedDict
+else:
+    from typing import NotRequired, TypedDict  # pylint: disable=ungrouped-imports
+
+LOGGER = logging.getLogger(__name__)
+
+BatchStatisticalRequestType = Union[str, dict, "BatchStatisticalRequest"]
+
+
+class S3Specification(TypedDict):
+    """Specification of a S3 path."""
+
+    url: str
+    accessKey: str
+    secretAccessKey: str
+    region: NotRequired[str]
+
+
+class AccessSpecification(TypedDict):
+    """Specification of a S3 input or output."""
+
+    s3: S3Specification
+
+
+class SentinelHubBatchStatistical(BaseBatchClient["BatchStatisticalRequest"]):
```

Comment:
Could we add a job monitor method similar to batch to the class? I have a code snippet in [cell 20 of the notebook](https://git.sinergise.com/sh-vas/technical-support/clients/-/blob/master/SuperAlberi/how_to_obtain_statistics_with_batch_statistical_api.ipynb), which update the progress bar by `completionPercentage`. An alternative way will be checking the files in the bucket and update by `files in the bucket / length of .gpkg file`

Reply:
Sorry... click the wrong option. Just wanted to add a comment... 🙈

Reply:
This will be added in a separate MR (so that we keep them a bit smaller and incremental)

Reply:
Thanks! May I ask if we plan to add other methods that extract statistics to a data frame from s3?